code/game/machinery/portable_turret.dm

```diff
@@ -102,6 +118,13 @@
 
 	//Sets up a spark system
 	spark_system = bind_spark(src, 5)
+	if(installation)
+		var/obj/item/weapon/gun/G = new installation(src)

```

Comment:
This does not look good, IMO.
```dm
		if(3)
			if(istype(I, /obj/item/weapon/gun/energy)) //the gun installation part
				E = I //typecasts the item to a gun
				if(E.can_turret)
					if(isrobot(user))
						return
					if(!user.unEquip(I))
						to_chat(user, "<span class='notice'>\the [I] is stuck to your hand, you cannot put it in \the [src]</span>")
						return
					to_chat(user, "<span class='notice'>You install [I] into the turret.</span>")
					user.drop_from_inventory(E,src)
					target_type = /obj/machinery/porta_turret
					installation = I.type //<-- gun is present here. You can get shot_delay here and hard code it otherwise.
					build_step = 4
					icon_state = "turret_frame_4_[case_sprite_set]"
					add_overlay("turret_[E.turret_sprite_set]_off")
					return
```

Reply:
Yeah, I know. I was gonna change that.